util/ranger/detacher.go

```diff
@@ -355,6 +355,112 @@ type DetachRangeResult struct {
 	IsDNFCond bool
 }
 
+// detachColumnsFromItem checks if the expression is one of GE,GT,LT,LE,EQ and NE function that one side is constant and
+// another is column or an in function which is `column in (constant list)`.
+// If so, it will return true. Otherwise, return false.
+func detachColumnsFromItem(expr expression.Expression, cols []*expression.Column) bool {
```

Comment:
Why do we have to do this check, in order to avoid expanding some unuseful DNF for range calculation?
For example, in this query `a and (b like 'xx' or b like 'zz') and (b = 2 or b > 5)`, will `(b like ...)` be skipped?

Reply:
Because `b like 'xx'` will convert to `eq(b, "xx")`, so it will be not skipped. But `b like 'xx%'` will be. There are two ways to solve. One is to deal with `b like 'xx%'`, the other is not to check.  If we assume that the order of the query is consistent with the order of the compound index, whether to check will not affect.